rasa/graph_components/adders/nlu_prediction_to_history_adder.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+import logging
+from rasa.shared.core.events import UserUttered
+from typing import Dict, Text, Any, List
+
+from rasa.core.channels.channel import UserMessage
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.nlu.constants import ENTITIES, INTENT, TEXT
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.core.trackers import DialogueStateTracker
+from rasa.shared.core.domain import Domain
+
+logger = logging.getLogger(__name__)
+
+
+class NLUPredictionToHistoryAdder(GraphComponent):
+    """Adds NLU predictions to DialogueStateTracker."""
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> NLUPredictionToHistoryAdder:
+        """Creates component (see parent class for full docstring)."""
+        return cls()
+
+    def add(
+        self,
+        predictions: List["Message"],
```

Comment:
Can we please not use forward references? We check the graph schema using the type annotations and forward references tend to break that
```suggestion
        predictions: List[Message],
```

Reply:
Gotcha, was not aware of that